### PR TITLE
API-4259: return errors from PPNS service and do not save subscription fields when 

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**
error occurs

### CODE REVIEW

File: app/uk/gov/hmrc/apisubscriptionfields/service/PushPullNotificationService.scala [hunk 1]
```diff
@@ -33,27 +31,27 @@ class PushPullNotificationService @Inject()(ppnsConnector: PushPullNotificationS
...
+  //     fieldDefinitions
+  //     .filter(_.`type` == FieldDefinitionType.PPNS_FIELD )
+  //     .map { fieldDefn =>
+  //       subscribeToPPNS(clientId, apiContext, apiVersion, fieldDefn, fields.get(fieldDefn.name).filterNot(_.isEmpty))
+  //     }
 
-    Future.sequence(subscriptionResponses).map(_ => ())
-  }
+  //   Future.sequence(subscriptionResponses).map(_ => ())
+  // }
```
Comment (peteslater-ee) [inline app/uk/gov/hmrc/apisubscriptionfields/service/PushPullNotificationService.scala:56]: Remove commented code

File: app/uk/gov/hmrc/apisubscriptionfields/connector/Model.scala [hunk 2]
```diff
@@ -27,3 +27,7 @@ private[connector] case class SubscriberRequest(callBackUrl: String, subscriberT
 private[connector] case class UpdateSubscriberRequest(subscriber: SubscriberRequest)
 
 private[connector] case class UpdateSubscriberResponse(boxId: BoxId)
+
+private[connector] case class UpdateCallBackUrlRequest(callBackUrl: String)
+
+private[connector] case class UpdateCallBackUrlResponse(successful: Boolean, errorMessage: Option[String])
```
Comment (peteslater-ee) [inline app/uk/gov/hmrc/apisubscriptionfields/connector/Model.scala:33]: Not sure about this - surely the response success or error can be determine by http status code and therefore a flag to say if it worked or not is not needed?

File: app/uk/gov/hmrc/apisubscriptionfields/service/PushPullNotificationService.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/apisubscriptionfields/service/PushPullNotificationService.scala:56]: Oops, will do!

File: app/uk/gov/hmrc/apisubscriptionfields/connector/Model.scala (hunk 2 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/apisubscriptionfields/connector/Model.scala:33]: We need to return the error to the UI.

File: app/uk/gov/hmrc/apisubscriptionfields/service/PushPullNotificationService.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/apisubscriptionfields/service/PushPullNotificationService.scala]: Done

File: app/uk/gov/hmrc/apisubscriptionfields/connector/Model.scala (hunk 2 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/apisubscriptionfields/connector/Model.scala:33]: Its actually the main point of the changes we have introduced

**Pull request merged**